### PR TITLE
 chore(refactor): #1229 move spinner to taps

### LINKED ISSUE #1229: Move spinner triggers to taps

### PR DESCRIPTION
**Pull request by eddiejaoude**
closes #1229 

### Notes

A summary of what was achieved in this PR

- [ ] moved spinner to pre/post `tap` - looks the same, starts slightly later and finishes the same time
- [ ] removed unused spinner imports

### CODE REVIEW

**Comment by eddiejaoude** [comment]
@webkhushboo please review when you get time 🤓

File: web/src/app/core/services/authentication.service.ts [hunk 1]
```diff
@@ -84,11 +84,11 @@ export class AuthenticationService {
 
     // This function used to login via github
     public login(): void {
-        this.spinnerService.setProgressBar(true);
         const provider: auth.GithubAuthProvider = new auth.GithubAuthProvider();
         provider.addScope('repo,admin:repo_hook');
         from(this.afAuth.auth.signInWithPopup(provider))
             .pipe(
+                tap(() => this.spinnerService.setProgressBar(true)),
```
Comment (webkhushboo) [inline web/src/app/core/services/authentication.service.ts]: Initially we were starting spinner , when we click on login button now after adding tap , its starting spinner once you enter sign in details in popup . Do you want this change intentionally ?

Comment (eddiejaoude) [reply web/src/app/core/services/authentication.service.ts]: My concern is, if someone closes the small popup window without signing in, the progress spinner keeps going and never ends (plus I think it is a bit cleaner code to have it in the `pipe`). So I think it makes more sense for the progress bar to start once the small popup closes after the user has signed in.

Comment (webkhushboo) [reply web/src/app/core/services/authentication.service.ts]: Thanks for clarification . Looks good 👍

**Pull request merged**